File: java/solitaire/deck/src/main/java/org/silnith/deck/Value.java
package org.silnith.deck;

/**
 * A numeric value for a playing card.
 * 
 * @author <a href="mailto:silnith@gmail.com">Kent Rosenkoetter</a>
 */
public enum Value {
    /**
     * The ace.  Usually the lowest card, but in some games is also the highest card.
     * Has a numeric value of {@code 1}.
     */
    ACE(1),
    /**
     * The value {@code 2}.
     */
    TWO(2),
    /**
     * The value {@code 3}.
     */
    THREE(3),
    /**
     * The value {@code 4}.
     */
    FOUR(4),
    /**
     * The value {@code 5}.
     */
    FIVE(5),
    /**
     * The value {@code 6}.
     */
    SIX(6),
    /**
     * The value {@code 7}.
     */
    SEVEN(7),
    /**
     * The value {@code 8}.
     */
    EIGHT(8),
    /**
     * The value {@code 9}.
     */
    NINE(9),
    /**
     * The value {@code 10}.
     */
    TEN(10),
    /**
     * The jack.  Has the value {@code 11}, though some games treat it as {@code 10}.
     */
    JACK(11),
    /**
     * The queen.  Has the value {@code 12}, though some games treat it as {@code 10}.
     */
    QUEEN(12),
    /**
     * The king.  Has the value {@code 13}, though some games treat it as {@code 10}.
     * The highest-value card, unless a game overrides this by treating the {@link #ACE ace}
     * as higher.
     */
    KING(13);
    
    private final int value;
    
    private Value(final int value) {
        this.value = value;
    }
    
    /**
     * Returns the numeric equivalent for the card value.
     * 
     * @return the numeric equivalent for the card value
     */
    public int getValue() {
        return value;
    }
    
    /**
     * Returns a short symbol representing the card value. This is usually a
     * single character, except for {@link #TEN ten}.
     * 
     * @return a short symbol representing the card value
     */
    public String toSymbol() {
        switch (this) {
        case ACE:
            return "A";
        case JACK:
            return "J";
        case QUEEN:
            return "Q";
        case KING:
            return "K";
        default:
            return String.valueOf(value);
        }
    }
    
    @Override
    public String toString() {
        switch (this) {
        case ACE:
            return "Ace";
        case TWO:
            return "Two";
        case THREE:
            return "Three";
        case FOUR:
            return "Four";
        case FIVE:
            return "Five";
        case SIX:
            return "Six";
        case SEVEN:
            return "Seven";
        case EIGHT:
            return "Eight";
        case NINE:
            return "Nine";
        case TEN:
            return "Ten";
        case JACK:
            return "Jack";
        case QUEEN:
            return "Queen";
        case KING:
            return "King";
        default:
            return super.toString();
        }
    }
    
}

File: csharp/Solitaire/Deck/Value.cs
using System;

namespace Silnith.Game.Deck
{
    /// <summary>
    /// A numeric value for a playing card.
    /// </summary>
    public enum Value
    {
        /// <summary>
        /// The ace.  Usually the lowest card, but in some games is also the highest card.
        /// Has a numeric value of <c>1</c>.
        /// </summary>
        Ace = 1,
        /// <summary>
        /// The value <c>2</c>.
        /// </summary>
        Two,
        /// <summary>
        /// The value <c>3</c>.
        /// </summary>
        Three,
        /// <summary>
        /// The value <c>4</c>.
        /// </summary>
        Four,
        /// <summary>
        /// The value <c>5</c>.
        /// </summary>
        Five,
        /// <summary>
        /// The value <c>6</c>.
        /// </summary>
        Six,
        /// <summary>
        /// The value <c>7</c>.
        /// </summary>
        Seven,
        /// <summary>
        /// The value <c>8</c>.
        /// </summary>
        Eight,
        /// <summary>
        /// The value <c>9</c>.
        /// </summary>
        Nine,
        /// <summary>
        /// The value <c>10</c>.
        /// </summary>
        Ten,
        /// <summary>
        /// The jack.  Has the value <c>11</c>, though some games treat it as <c>10</c>.
        /// </summary>
        Jack,
        /// <summary>
        /// The queen.  Has the value <c>12</c>, though some games threat it as <c>10</c>.
        /// </summary>
        Queen,
        /// <summary>
        /// The king.  Has the value <c>13</c>, though some games treat it as <c>10</c>.
        /// The highest-value card, unless a game overrides this by treating the <see cref="Ace"/>
        /// as higher.
        /// </summary>
        King
    }

    /// <summary>
    /// Extension methods for <see cref="Value"/>.
    /// </summary>
    public static class ValueExtensions
    {
        /// <summary>
        /// Returns the numeric equivalent for the card value.
        /// </summary>
        /// <param name="value">The card value.</param>
        /// <returns>The numeric equivalent for the card value.</returns>
        public static int GetValue(this Value value)
        {
            return (int)value;
        }

        /// <summary>
        /// Returns a short symbol representing the card value.
        /// This is usually a single character, except for <see cref="Value.Ten"/>.
        /// </summary>
        /// <param name="value">The card value.</param>
        /// <returns>A short symbol representing the card value.</returns>
        public static string ToSymbol(this Value value)
        {
            return value switch
            {
                Value.Ace => "A",
                Value.Jack => "J",
                Value.Queen => "Q",
                Value.King => "K",
                _ => $"{value:D}",
            };
        }
    }
}
